C2GUILauncher/src/Chivalry2Launcher.cs

```diff
@@ -40,21 +40,29 @@ public class Chivalry2Launcher {
             return VanillaLauncher.Launch(string.Join(" ", args));
         }
 
-        public Thread? LaunchModded(Window window, InstallationType installationType, List<string> args, bool downloadPlugin, bool enablePluginLogging, Process? serverRegister = null) {
+        public async Task<Thread?> LaunchModded(Window window, InstallationType installationType, List<string> args, bool downloadPlugin, bool enablePluginLogging, Process? serverRegister = null) {
             if (installationType == InstallationType.NotSet) return null;
 
             logger.Info("Attempting to launch modded game.");
 
+            if (!this.ModManager.EnabledModReleases.Any(x => x.Manifest.RepoUrl.EndsWith("Chiv2-Community/Unchained-Mods"))) {
+                logger.Info("Unchained-Mods mod not enabled. Enabling.");
+                var hasModList = this.ModManager.Mods.Any();
+
+                if (!hasModList)
+                    await this.ModManager.UpdateModsList();
```

Comment:
This call has to happen on the UI thread because its updating an ObservableCollection, so I moved this piece out of the other thread.